src/lib.rs

```diff
@@ -271,6 +271,62 @@ impl PolicyServer {
     }
 }
 
+// Load the ServerConfig to be used by the Policy Server configuring the server
+// certificate and mTLS when necessary
+//
+// RustlsConfig does not offer a function to load the client CA certificate together with the
+// service certificates. Therefore, we need to load everything and build the ServerConfig
+async fn build_tls_server_config(tls_config: &TlsConfig) -> Result<rustls::ServerConfig> {
+    use std::{fs::File, io::BufReader};
+
+    use rustls::{server::WebPkiClientVerifier, RootCertStore, ServerConfig};
+    use rustls_pemfile::Item;
+    use rustls_pki_types::{CertificateDer, PrivateKeyDer};
+
+    let cert_file = &mut BufReader::new(File::open(tls_config.cert_file.clone()).unwrap());
+    let key_file = &mut BufReader::new(File::open(tls_config.key_file.clone()).unwrap());
+    let cert = rustls_pemfile::certs(cert_file)
+        .map(|it| it.map(|it| it.to_vec()))
+        .collect::<Result<Vec<_>, _>>()?;
+    let mut key_vec: Vec<Vec<u8>> = rustls_pemfile::read_all(key_file)
+        .filter_map(|i| match i.ok()? {
+            Item::Sec1Key(key) => Some(key.secret_sec1_der().to_vec()),
+            Item::Pkcs1Key(key) => Some(key.secret_pkcs1_der().to_vec()),
+            Item::Pkcs8Key(key) => Some(key.secret_pkcs8_der().to_vec()),
+            _ => None,
```

Comment:
I would log that, it's odd to put a something that is not a key into a pem file that is supposed to contain a key:

```rust
_ => {
   info!("found something that is not a key, ignoring");
   None
}
```

Obviously with a better message :)